src/hooks/useDesktopNotifications.ts

```diff
@@ -0,0 +1,144 @@
+            )
+        )
+        .filter((pr) => pr.pr.user?.login !== login);
+      for (const newPr of newPrsReadyForReview) {
+        notify(
+          'PR ready',
+          `${newPr.pr.title} is ready for review`,
+          newPr.pr.html_url,
+          newPr.pr.user?.avatar_url
+        );
+      }
+
+      // Notify if any of our PRs have new reviews
+      const currentReviews = getReviews(rawData, login);
+      const previousReviews = getReviews(previousData, login);
+      const newReviews = currentReviews.filter(
+        (r) => !previousReviews.some((prevR) => prevR.id === r.id)
+      );
+      for (const review of newReviews) {
+        notify(
+          review.state,
+          `${review.user?.login} reviewed: ${review.body}`,
+          review.html_url,
+          review.user?.avatar_url
+        );
+      }
+
+      if (notifyForComments) {
+        // Notify if any of our PRs have new comments
+        const currentComments = getComments(rawData, login);
+        const previousComments = getComments(previousData, login);
+        const newComments = currentComments.filter(
+          (c) => !previousComments.some((prevC) => prevC.id === c.id)
+        );
+        for (const comment of newComments) {
+          notify(
+            'New comment',
+            `${comment.user?.login} commented: ${comment.body}`,
+            comment.html_url,
+            comment.user?.avatar_url
+          );
+        }
+      }
+    };
+
+    notifyOfUpdates();
+
+    setPreviousData(rawData);
+  }, [rawData, notificationsEnabled, previousData, login, notifyForComments]);
+
+  const notify = (
+    title: string,
+    body: string,
+    clickUrl: string,
+    iconUrl = '/logo192.png'
+  ) => {
+    const notification = new Notification(title, {
+      body,
+      icon: iconUrl,
+    });
+    notification.onclick = () => {
+      // window.focus();
+      window.open(clickUrl, '_blank')?.opener?.focus();
+      notification.close();
+    };
+  };
+}
+
+const getPrsReadyForReview = (data: PrData[], login: string) =>
+  data.filter(
+    (pr) =>
+      pr.pr.state === 'open' &&
+      !pr.pr.draft &&
+      !pr.pr.merged_at &&
+      (!Object.values(pr.reviews ?? {}).some(
+        (r) => r?.state === 'CHANGES_REQUESTED'
+      ) ||
+        pr.pr.requested_reviewers?.some((r) => r.login === login))
+  );
+
+const getComments = (data: PrData[], login: string) => {
+  let comments = [];
+  for (const pr of data) {
+    if (pr.pr.state !== 'open' || !!pr.pr.merged_at) continue;
+
+    // If it's not our pr only get comments in reply to our reviews or ones we're tagged in
+    let prComments = pr.comments?.filter((c) => c.user?.login !== login);
+    if (pr.pr.user?.login !== login)
+      prComments = prComments?.filter(
+        (c) => c.in_reply_to_id || c.body.includes(`@${login}`)
+      );
+
+    if (prComments?.length) comments.push(prComments);
+  }
+  return comments.flat();
+};
```

Comment:
I think we should just be notifying for comments where:
- you're the author, or
- you've been requested for review, or
- you've provided a review

atm this looks like it won't notify you for PR comments on PRs you're reviewing unless it tags you/is in response to you

Reply:
Additionally, this can be cleaned up a bit. Instead of using `push` and `flat`, a `reduce` would be cleaner.

```tsx
const getComments = (data: PrData[], login: string) => {
  const filteredData = data.filter((pr) => pr.pr.state === 'open' && !pr.pr.merged_at && pr.comments);

  return filteredData.reduce((comments, pr) => {
    // If it's not our pr only get comments in reply to our reviews or ones we're tagged in
    const prComments = pr.comments.filter((c) => c.user?.login !== login);
    if (pr.pr.user?.login === login) {
      // my pr, notify for all comments
      return [...comments, ...prComments];
    }

    // not my PR, filter to relevant comments
    return [...comments, ...prComments.filter((c) => c.in_reply_to_id || c.body.includes(`@${login}`))];
  }, []);
```